### PR TITLE
fix(ios): allow pods mixte type settings on post-install

### PR DESCRIPTION
**Pull request by MasGaNo**
## Summary:

Following the discussion on https://github.com/facebook/react-native/issues/46505, this PR aims to allow mixte type configuration (String and/or Array of String) during the post installation of pods.

## Changelog:

[IOS] [FIXED] - allow pods mixte type settings on post-install

## Test Plan:

`packages/react-native/scripts/cocoapods/__tests__/utils-test.rb` test suits was updated to support array and works as expected

### CODE REVIEW

**Review by cipolleschi — APPROVED** [review]
Thanks for the contribution! 😄

**Comment by cipolleschi** [comment]
@MasGaNo could you sign the CLA? otherwise I can't import it in the internal repository

**Comment by MasGaNo** [comment]
Hey @cipolleschi 

I already signed it 😄 

I was waiting for approval, seems to be fine now ^^

**Pull request closed without merging**
